container.go

```diff
@@ -482,6 +482,9 @@ type HostConfig struct {
 //
 // See http://goo.gl/iM5GYs for more details.
 func (c *Client) StartContainer(id string, hostConfig *HostConfig) error {
+	if id == "" {
+		return &NoSuchContainer{ID: id, Err: fmt.Errorf("empty container id")}
```

Comment:
I think this should be a new error type instead of using NoSuchContainer - NoSuchContainer is used to denote when you actually have an invalid container id, just just had an empty parameter string.